data/transactions/logic/evalStateful_test.go

```diff
@@ -466,6 +468,18 @@ func (l *testLedger) GetDelta(txn *transactions.Transaction) (evalDelta basics.E
 	return
 }
 
+func (l *testLedger) AppendLog(value string) error {
+	appIdx := l.appID
+	_, ok := l.applications[appIdx]
+	if !ok {
+		return fmt.Errorf("no such app")
+	}
+
+	// append logs
+	l.logs[appIdx] = append(l.logs[appIdx], value)
```

Comment:
If this is the only way a log can be written, the l.logs should probably just be []string, not a map.  Only the current app can have logs.